test/e2e/apimachinery/watch.go

```diff
@@ -381,3 +426,64 @@ func waitForEvent(w watch.Interface, expectType watch.EventType, expectObject ru
 		}
 	}
 }
+
+func waitForNextConfigMapEvent(watch watch.Interface) *v1.ConfigMap {
+	select {
+	case event := <-watch.ResultChan():
+		if configMap, ok := event.Object.(*v1.ConfigMap); ok {
+			return configMap
+		} else {
+			framework.Failf("expected config map")
+		}
+	case <-time.After(10 * time.Second):
+		framework.Failf("timed out waiting for watch event")
+	}
+	return nil // should never happen
+}
+
+const (
+	createEvent = 0
+	updateEvent = iota
+	deleteEvent = iota
+)
+
+func produceConfigMapEvents(f *framework.Framework, stopc <-chan struct{}, minWaitBetweenEvents time.Duration) {
+	c := f.ClientSet
+	ns := f.Namespace.Name
+
+	existing := []int{}
+	for i := 0; ; i++ {
+		waitc := time.After(minWaitBetweenEvents) // rate limit
+		op := rand.Int() % 2
+		if len(existing) == 0 {
+			op = createEvent
+		}
+
+		cm := &v1.ConfigMap{}
+		switch op {
+		case createEvent:
+			cm.Name = fmt.Sprintf("cm-%d", i)
+			_, err := c.CoreV1().ConfigMaps(ns).Create(cm)
+			Expect(err).NotTo(HaveOccurred())
+			existing = append(existing, i)
+		case updateEvent:
+			idx := rand.Int() % len(existing)
+			cm.Name = fmt.Sprintf("cm-%d", existing[idx])
+			_, err := c.CoreV1().ConfigMaps(ns).Update(cm)
+			Expect(err).NotTo(HaveOccurred())
+		case deleteEvent:
+			idx := rand.Int() % len(existing)
+			name := fmt.Sprintf("cm-%d", existing[idx])
```

Comment:
nit: might be worth it to declare `name := func(n int) string { return fmt.Sprintf("cm-%d", existing[n]) }` somewhere?

Reply:
sg